### PR TITLE
Bump ubi 9.4

### PR DESCRIPTION
**Pull request by jotak**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,10 +26,10 @@ RUN GOARCH=$TARGETARCH make compile
 RUN mkdir -p output
 
 # Create final image from ubi + built binary
-FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9/ubi:9.3
+FROM --platform=$TARGETPLATFORM registry.access.redhat.com/ubi9/ubi:9.4
```
Comment (jotak) [inline Dockerfile:29]: guys, is there a reason here to use the "full" ubi and not the minimal like we do elsewhere ? Is it that we need more tooling in the image?

Comment (msherif1234) [reply Dockerfile:29]: I just tried building running e2e using `ubi-minimal:9.4` it shows no issue I don't think we need the full image here you switched in this PR

**Comment by jotak** [comment]
@jpinsonneau , looking at the error, it seems like in the e2e test "TestFlowCapture", the output/flow directory does not exist for some reason, when using a minimal base image. Any idea why?

konflux bot is also upgrading the ubi to 9.4: https://github.com/netobserv/network-observability-cli/pull/70 - so if we think this is good enough and we don't need the minimal image, I can close this PR. But I'd be curious to understand what makes it fail with minimal

**Comment by jpinsonneau** [comment]
> @jpinsonneau , looking at the error, it seems like in the e2e test "TestFlowCapture", the output/flow directory does not exist for some reason, when using a minimal base image. Any idea why?
> 
> konflux bot is also upgrading the ubi to 9.4: #70 - so if we think this is good enough and we don't need the minimal image, I can close this PR. But I'd be curious to understand what makes it fail with minimal

UBI 9.3 contains `package tar-1.34-6.el9_1.x86_64` GNU file archiving program
which is not present in minimal container image.

As discussed in [this slack thread](https://redhat-internal.slack.com/archives/C02939DP5L5/p1721718492868999?thread_ts=1721718022.672209&cid=C02939DP5L5), you need to have `tar` installed on the container side to properly run `oc cp`.

Did you try the `rsync` alternative ?

**Comment by jotak** [comment]
let's see if adding tar fixes it then (I prefer to stick with the current behaviour as much as possible, and adding tar was something we do downstream also)

**Comment by jotak** [comment]
@jpinsonneau e2e pass now \o/

**Review by jpinsonneau — APPROVED** [review]
Code looks good, thanks @jotak !

**Comment by jotak** [comment]
@jpinsonneau I think [this change](https://github.com/openshift/release/pull/55030) will be necessary for the prow/image test to pass

**Pull request merged**